platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDb.java

```diff
@@ -205,6 +212,26 @@ private MerkleDb(final Path storageDir) {
         storeMetadata();
     }
 
+    /**
+     * Iterates over the list of table metadata starting from index from {@link #nextTableId} until
+     * it finds an ID that isn't used by a table.
+     *
+     * @return The next available table ID
+     */
+    private int getNextTableId() {
+        int tablesCount = 0;
+        while (true) {
```

Comment:
Instead of an infinite loop with returns, you might consider a loop with explicit end.
```suggestion
            for(tablesCount = 0; tablesCount <= MAX_TABLES ; tablesCount++) {
                final int id = Math.abs(nextTableId.getAndIncrement() % MAX_TABLES);
                if (tableConfigs.get(id) == null) {
                    return id;
                }
            }
            throw new IllegalStateException("Tables limit is reached");
```

Reply:
Good idea. I'll rewrite it this way.